state/user.go

```diff
@@ -129,6 +141,65 @@ func (st *State) addUser(name, displayName, password, creator string, secretKey
 	return user, nil
 }
 
+// updateExistingUser manipulates the values of an existing user in the db.
+// This is particularly useful when reusing existing users that were previously
+// deleted.
+func (st *State) updateExistingUser(u *User, displayName, password, creator string, secretKey []byte) (*User, error) {
+
+	dateCreated := st.nowToTheSecond()
```

Comment:
if we overwrite the creation date, then there is no way of auditing previously "deleted" users, in that case why not simply really delete the user's doc when removing a user?